src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/DeleteWalletByIdIT.java

```diff
@@ -0,0 +1,66 @@
+package pl.byczazagroda.trackexpensesappbackend.integration;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import pl.byczazagroda.trackexpensesappbackend.BaseIntegrationTestIT;
+import pl.byczazagroda.trackexpensesappbackend.exception.ErrorCode;
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransaction;
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransactionType;
+import pl.byczazagroda.trackexpensesappbackend.model.Wallet;
+import pl.byczazagroda.trackexpensesappbackend.repository.FinancialTransactionRepository;
+import pl.byczazagroda.trackexpensesappbackend.repository.WalletRepository;
+
+import java.math.BigDecimal;
+import java.time.Instant;
+
+class DeleteWalletByIdIT extends BaseIntegrationTestIT {
+
+    @Autowired
+    WalletRepository walletRepository;
+
+    @Autowired
+    FinancialTransactionRepository financialTransactionRepository;
+
+    @BeforeEach
+    void clearDatabase() {
+        walletRepository.deleteAll();
+    }
+
+    @DisplayName("Should delete wallet from a database and return status 'OK'")
+    @Test
+    void testDeleteWalletByIdAPI_whenWalletIdIsCorrect_thenShouldReturnAcceptAndDeleteRecord() throws Exception {
+        Wallet wallet = walletRepository.save(new Wallet("Test Wallet"));
+        createTestFinancialTransaction(wallet);
+        mockMvc.perform(MockMvcRequestBuilders.delete("/api/wallets/{id}", wallet.getId()).accept(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.status().isOk());
+        Assertions.assertEquals(0, walletRepository.count());
+    }
```

Comment:
When we delete a wallet, we also delete all related transactions. Add a check that  transactions are also deleted.
`Assertions.assertEquals(0, financialTransactionRepository.count());`

Reply:
You should add `Assertions.assertEquals(0, financialTransactionRepository.count());`  between line 42-43 too.